src/app/lib/analyticsUtils/index.js

```diff
@@ -333,4 +333,272 @@ export const getCampaignType = () => {
+      value: 'AD',
+      wrap: false,
+    },
+    {
+      description: 'at_campaign field',
+      value: getMarketingUrlParam(query, 'at_campaign'),
+      wrap: false,
+    },
+    {
+      description: 'at_creation field',
+      value: getMarketingUrlParam(query, 'at_creation'),
+      wrap: true,
+    },
+    {
+      description: 'at_variant field',
+      value: getMarketingUrlParam(query, 'at_variant'),
+      wrap: true,
+    },
+    {
+      description: 'at_format field',
+      value: getMarketingUrlParam(query, 'at_format'),
+      wrap: true,
+    },
+    {
+      description: 'blank value (-)',
+      value: '',
+      wrap: false,
+    },
+    {
+      description: 'at_general_placement field',
+      value: getMarketingUrlParam(query, 'at_general_placement'),
+      wrap: true,
+    },
+    {
+      description: 'at_detail_placement field',
+      value: getMarketingUrlParam(query, 'at_detail_placement'),
+      wrap: true,
+    },
+  ]);
+};
+
+export const getCustomMarketingString = href => {
+  const { query } = new Url(href, true);
+  return buildMarketingString([
+    {
+      description: 'custom campaign prefix',
+      value: `CS${getMarketingUrlParam(query, 'at_medium').replace(
+        'custom',
+        '',
+      )}`,
+      wrap: false,
+    },
+    {
+      description: 'at_campaign field',
+      value: getMarketingUrlParam(query, 'at_campaign'),
+      wrap: false,
+    },
+    {
+      description: 'at_custom1 field',
+      value: getMarketingUrlParam(query, 'at_custom1'),
+      wrap: true,
+    },
+    {
+      description: 'at_custom2 field',
+      value: getMarketingUrlParam(query, 'at_custom2'),
+      wrap: true,
+    },
+    {
+      description: 'at_custom_3 field',
+      value: getMarketingUrlParam(query, 'at_custom3'),
+      wrap: true,
+    },
+    {
+      description: 'at_custom_4 field',
+      value: getMarketingUrlParam(query, 'at_custom4'),
+      wrap: true,
+    },
+  ]);
+};
+
+/* This transforms urls with a hash param to query params
+   ie. from bbc.com/mundo#some_param_1=24&some_param_2=48 to bbc.com/mundo?some_param_1=24&some_param_2=48
+*/
+const parameteriseHash = hash => new Url(hash.replace('#', '?'), true).query;
+
+export const getXtorMarketingString = href => {
+  const { query, hash } = new Url(href, true);
+  const field = 'xtor';
+
+  const marketingString =
+    getMarketingUrlParam(query, field) ||
+    getMarketingUrlParam(parameteriseHash(hash), field);
+
+  return marketingString || null;
+};
+
+export const getATIMarketingString = (href, campaignType) => {
+  if (!campaignType) return null;
+
+  const supportedCampaignMappings = {
+    affiliate: () => getAffiliateMarketingString(href),
+    sl: () => getSLMarketingString(href),
+    email: () => getEmailMarketingString(href),
+    display: () => getDisplayMarketingString(href),
+    custom: () => getCustomMarketingString(href),
+    XTOR: () => getXtorMarketingString(href),
+  };
```

Comment:
I don't know if you want to add a try-catch around these calls, there is a lot of specific logic for each kind of marketing string and there is a possibility an error could be thrown, I don't know what would happen if we did throw in here, we don't have error boundaries in our app so not quite sure. Maybe at least test throwing an error from here an see what happens!

Reply:
@andrewscfc [there's this bit](https://github.com/bbc/simorgh/pull/7429/files#diff-1c3985090b52f7b568ce9f3651bd02abR599-R601) where this is null checked so that doesn't happen eventually

Reply:
I was thinking if we had any cases that throw errors, this bit made me suggest it: https://github.com/bbc/simorgh/pull/7429/files#diff-1c3985090b52f7b568ce9f3651bd02abR439-R443 

If the value wasn't `aquisition`, `retention` or `promotion` it would return `null`. That though would probably just be inserted into the request string as `null-`. There probably isn't anything that throws an error so should be okay.